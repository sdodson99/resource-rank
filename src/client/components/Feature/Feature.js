import React from 'react';
import PropTypes from 'prop-types';
import styles from './Feature.module.css';

const Feature = ({ feature }) => {
  const { title, description, imageSrc } = feature;

  return (
    <div className={styles.Feature} data-testid="Feature">
      <div>
        <img className={styles.Image} src={imageSrc} alt={title} />
      </div>
      <div className={styles.Title}>
        <strong>{title}</strong>
      </div>
      <div className={styles.Description}>{description}</div>
    </div>
  );
};

Feature.propTypes = {
  feature: PropTypes.shape({
    title: PropTypes.node.isRequired,
    description: PropTypes.node.isRequired,
    imageSrc: PropTypes.string.isRequired,
  }).isRequired,
};

Feature.defaultProps = {};

export default Feature;
