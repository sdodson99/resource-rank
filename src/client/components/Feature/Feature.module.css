.Feature {
  @apply sm:flex flex-grow items-center;
}

.FeatureDescription {
  @apply flex-1 text-center sm:text-left text-3xl;
}

.FeatureImage {
  max-height: 150px;
  max-width: 150px;

  @apply sm:ml-10 mx-auto mt-8 sm:mt-0;
}
