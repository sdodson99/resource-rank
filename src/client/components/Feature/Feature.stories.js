import React from 'react';
import Feature from './Feature';

export default {
  component: Feature,
  title: 'Components/Feature',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <Feature {...args} />;

export const Default = Template.bind({});
Default.args = {
  feature: {
    title: 'Super Fast!',
    description: 'We do our best to make things fast.',
    imageSrc: '/img/instantly-find.svg',
  },
};
