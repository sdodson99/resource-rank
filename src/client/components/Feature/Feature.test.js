import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Feature from './Feature';

describe('<Feature />', () => {
  test('it should mount', () => {
    render(<Feature />);

    const feature = screen.getByTestId('Feature');

    expect(feature).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<Feature description="123" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
