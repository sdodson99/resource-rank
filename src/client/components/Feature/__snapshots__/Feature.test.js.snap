// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> should render correctly 1`] = `
<div
  className="Feature"
  data-testid="Feature"
>
  <div>
    <img
      alt="Title"
      className="Image"
      src="ImageSrc"
    />
  </div>
  <div
    className="Title"
  >
    <strong>
      Title
    </strong>
  </div>
  <div
    className="Description"
  >
    Description
  </div>
</div>
`;
